### PR TITLE
feat: [sc-26579] Develop spacelift-stack Terraform Module

### PR DESCRIPTION
**Pull request by Cwagne17**
Story details: https://app.shortcut.com/clarkcan/story/26579

### CODE REVIEW

File: modules/spacelift-stack/README.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
+# Spacelift Stack
+
+## Overview
+
+This module is designed to create a stack and its supporting resources, enabling an opinionated Continuous Integration/Continuous Deployment (CICD) workflow on Spacelift. It provides the capability to create a Spacelift stack and optionally set up an IAM role for integration with AWS.
+
+![Spacelift Diagram](../../_docs/tf-aws-spacelift-stack.png)
```
Comment (mfranz1) [inline modules/spacelift-stack/README.md:7]: This image isn't loading in the rich diff

File: modules/spacelift-stack/README.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
+# Spacelift Stack
+
+## Overview
+
+This module is designed to create a stack and its supporting resources, enabling an opinionated Continuous Integration/Continuous Deployment (CICD) workflow on Spacelift. It provides the capability to create a Spacelift stack and optionally set up an IAM role for integration with AWS.
```
Comment (mfranz1) [inline modules/spacelift-stack/README.md:5]: Opinionated?

File: modules/spacelift-stack/README.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+Description: n/a
+
+### <a name="output_stack_iam_role_policy_arns"></a> [stack\_iam\_role\_policy\_arns](#output\_stack\_iam\_role\_policy\_arns)
+
+Description: n/a
+
+### <a name="output_stack_id"></a> [stack\_id](#output\_stack\_id)
+
+Description: n/a
```
Comment (mfranz1) [inline modules/spacelift-stack/README.md:327]: Should these have descriptions?

File: modules/spacelift-stack/main.tf [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+  path = local.iam_role_path
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      jsondecode(data.spacelift_aws_integration_attachment_external_id.this[0].assume_role_policy_statement),
+    ]
+  })
+  force_detach_policies = true
+}
```
Comment (mfranz1) [inline modules/spacelift-stack/main.tf:283]: Trash Chris. Create the policy document!

**Review by JosueMolinaMorales — APPROVED** [review]

File: examples/deploy-ecs-cluster/variables.tf [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+# OPTIONAL PARAMETERS
+
+# These values are optional and have default values
+
+# --------------------------------------------------------------------
+
+variable "cluster_instance_ami" {
+  type        = string
+  description = "The AMI to run on each instance in the ECS cluster."
+  default     = "ami-0e692fe1bae5ca24c"
```
Comment (mfranz1) [inline examples/deploy-ecs-cluster/variables.tf:23]:
```suggestion
  default     = "ami-0e692fe1bae5ca24c"
```
is this supposed to be defaulted

Comment (mfranz1) [reply examples/deploy-ecs-cluster/variables.tf:23]: Pls ignore. I'm so sorry

**Review by mfranz1 — APPROVED** [review]

**Review by bereketsega — APPROVED** [review]
Looks good.

**Pull request merged**